pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -455,6 +441,31 @@ func hasClearedOutPreviouslyDefinedIngressList(oldObject, newObject *hivev1.Clus
 	return false
 }
 
+func validateIngressDomainsShareClusterDomain(newObject *hivev1.ClusterDeploymentSpec) bool {
+	// ingress entries must share the same domain as the cluster
+	regexString := fmt.Sprintf("(?i).*%s.%s$", newObject.ClusterName, newObject.BaseDomain)
```

Comment:
what is the purpose of the ?i (for my own education)... also, shouldn't the regexp include a period before the cluster name?

Reply:
case insenstive.
yes, adding the period will improve the pattern matching. will add that.

Reply:
added the leading `.` as part of the shared domain validation